File: frontend/src/api/product.ts
import { apiClient } from './client';
import { Product } from '../types';

export const createProduct = async (product: Omit<Product, 'id' | 'created_at'>): Promise<{ id: number; message: string }> => {
  const response = await apiClient.post('/api/product/create', {
    name: product.name,
    category: product.category || 'Consumer Brand',
    description: product.description,
    features: Array.isArray(product.features) ? product.features.join(', ') : product.features || '',
    problem: product.problem_solved || '',
    audience: product.target_audience,
    price_range: product.price || '',
    competitors: '',
    platform: product.platform,
    tone: product.tone,
    requirements: product.requirements || '',
  });
  return {
    id: response.data.product_id,
    message: response.data.message,
  };
};

export const getProduct = async (id: number): Promise<Product> => {
  const response = await apiClient.get(`/api/product/${id}`);
  return {
    id: response.data.id,
    name: response.data.name,
    category: response.data.category,
    description: response.data.description,
    features: response.data.features ? String(response.data.features).split(',').map((item) => item.trim()).filter(Boolean) : [],
    problem_solved: response.data.problem,
    target_audience: response.data.audience,
    price: response.data.price_range,
    platform: response.data.platform,
    tone: response.data.tone,
    requirements: response.data.requirements,
    created_at: response.data.created_at,
  };
};

File: backend/api/product.py
from fastapi import APIRouter, HTTPException

from database.database import execute, fetch_one
from schemas.product import ProductCreate, ProductOut, ProductResponse


router = APIRouter(prefix="/api/product", tags=["product"])


@router.post("/create", response_model=ProductResponse)
def create_product(payload: ProductCreate):
    product_id = execute(
        """
        INSERT INTO products (
            name, category, description, features, problem, audience,
            price_range, competitors, platform, tone, requirements
        )
        VALUES (?, ?, ?, ?, ?, ?, ?, ?, ?, ?, ?)
        """,
        (
            payload.name,
            payload.category,
            payload.description,
            payload.features,
            payload.problem,
            payload.audience,
            payload.price_range,
            payload.competitors,
            payload.platform,
            payload.tone,
            payload.requirements,
        ),
    )
    return {"product_id": product_id, "message": "Product saved successfully"}


@router.get("/{product_id}", response_model=ProductOut)
def get_product(product_id: int):
    row = fetch_one("SELECT * FROM products WHERE id = ?", (product_id,))
    if not row:
        raise HTTPException(status_code=404, detail="Product not found")
    return dict(row)
